bindings/muen/muen-net.c

```diff
@@ -103,62 +145,80 @@ void generate_mac_addr(uint8_t *addr)
+        return false;
     }
     if (!(chan_out->data.mem.flags & MEM_CHANNEL_FLAG)) {
-        log(WARN, "Solo5: Net: Memory '%s' is not a channel\n",
-            chan_out->name.data);
-        return;
+        log(WARN, "Solo5: Net: '%s': Memory '%s' is not a channel\n",
+            name, chan_out->name.data);
+        return false;
     }
     if (!(chan_out->data.mem.flags & MEM_WRITABLE_FLAG)) {
-        log(WARN, "Solo5: Net: Output channel '%s' not writable\n",
-            chan_out->name.data);
-        return;
+        log(WARN, "Solo5: Net: '%s': Output channel '%s' not writable\n",
+            name, chan_out->name.data);
+        return false;
     }
 
-    net_out = (struct muchannel *)(chan_out->data.mem.address);
-    muen_channel_init_writer(net_out, MUENNET_PROTO, sizeof(struct net_msg),
+    device->net_out = (struct muchannel *)(chan_out->data.mem.address);
+    muen_channel_init_writer(device->net_out, MUENNET_PROTO, sizeof(struct net_msg),
                              chan_out->data.mem.size, epoch);
-    log(INFO, "Solo5: Net: Muen shared memory stream, protocol 0x%llx\n",
-        MUENNET_PROTO);
-    log(INFO, "Solo5: Net: Output channel @ 0x%llx, size 0x%llx, epoch 0x%llx\n",
+    log(INFO, "Solo5: Net: '%s': Muen shared memory stream, protocol 0x%llx\n",
+        name, MUENNET_PROTO);
+    log(INFO, "Solo5: Net: '%s': Output channel @ 0x%llx, size 0x%llx, epoch 0x%llx\n",
+        name,
         (unsigned long long)chan_out->data.mem.address,
         (unsigned long long)chan_out->data.mem.size, (unsigned long long)epoch);
 
     if (!chan_in) {
-        log(WARN, "Solo5: Net: No input channel\n");
-        return;
+        log(WARN, "Solo5: Net: '%s': No input channel\n", name);
+        return false;
     }
     if (!(chan_in->data.mem.flags & MEM_CHANNEL_FLAG)) {
-        log(WARN, "Solo5: Net: Memory '%s' is not a channel\n",
-            chan_in->name.data);
-        return;
+        log(WARN, "Solo5: Net: '%s': Memory '%s' is not a channel\n",
+            name, chan_in->name.data);
+        return false;
     }
     if (chan_in->data.mem.flags & MEM_WRITABLE_FLAG) {
-        log(DEBUG, "Solo5: Net: Input channel '%s' is writable\n",
-            chan_in->name.data);
+        log(DEBUG, "Solo5: Net: '%s': Input channel '%s' is writable\n",
+            name, chan_in->name.data);
     }
 
-    net_in = (struct muchannel *)(chan_in->data.mem.address);
-    muen_channel_init_reader(&net_rdr, MUENNET_PROTO);
-    log(INFO, "Solo5: Net: Input  channel @ 0x%llx, size 0x%llx\n",
+    device->net_in = (struct muchannel *)(chan_in->data.mem.address);
+    muen_channel_init_reader(&device->net_rdr, MUENNET_PROTO);
+    log(INFO, "Solo5: Net: '%s': Input  channel @ 0x%llx, size 0x%llx\n",
+        name,
         (unsigned long long)chan_in->data.mem.address,
         (unsigned long long)chan_in->data.mem.size);
 
-    generate_mac_addr(mac_addr);
+    generate_mac_addr(device->mac_addr);
```

Comment:
I would have expected `&device->mac_addr` here since `device { uint8_t mac_addr[6]; }`, and `generate_mac_addr(uint8_t *)`, but maybe that's just me not understanding C :)

Reply:
@cfcs In C a `char []` is the same as a `char *`, so the code is correct.